certora/munged/protocol/libraries/types/DataTypes.sol

```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+library DataTypes {
+  struct ReserveData {
+    //stores the reserve configuration
+    ReserveConfigurationMap configuration;
+    //the liquidity index. Expressed in ray
+    uint128 liquidityIndex;
+    //the current supply rate. Expressed in ray
+    uint128 currentLiquidityRate;
+    //variable borrow index. Expressed in ray
+    uint128 variableBorrowIndex;
+    //the current variable borrow rate. Expressed in ray
+    uint128 currentVariableBorrowRate;
+    //the current stable borrow rate. Expressed in ray
+    uint128 currentStableBorrowRate;
+    //timestamp of last update
+    uint40 lastUpdateTimestamp;
+    //the id of the reserve. Represents the position in the list of the active reserves
+    uint16 id;
+    //aToken address
+    address aTokenAddress;
+    //stableDebtToken address
+    address stableDebtTokenAddress;
+    //variableDebtToken address
+    address variableDebtTokenAddress;
+    //address of the interest rate strategy
+    address interestRateStrategyAddress;
+    //the current treasury balance, scaled
+    uint128 accruedToTreasury;
+    //the outstanding unbacked aTokens minted through the bridging feature
+    uint128 unbacked;
+    //the outstanding debt borrowed against this asset in isolation mode
+    uint128 isolationModeTotalDebt;
+  }
+
+struct ReserveConfigurationMap {
+    uint256 data; //original data
+    // splitting to two object to avoid "Stack too deep error"
+    // for having more than 16 local variables in the same scope.
+    ReserveConfigurationMap_ints intData;
+    ReserveConfigurationMap_bools boolData;
+  }
+
+  struct ReserveConfigurationMap_ints {
```

Comment:
Why do we need to split `ReserveConfigurationMap` to two? (question mostly from curiosity :)

Reply:
Apparently, there's a limit on the number of variables that can be declared in the single scope. If it's more than 16, solc gives me `Stack too deep error` :) Adding a layer of `struct` solves it.